Storage/MockLogins.swift

```diff
@@ -34,6 +34,36 @@ public class MockLogins: BrowserLogins, SyncableLogins {
         return Deferred(value: Maybe(success: cursor))
     }
 
+    public func getAllLogins() -> Deferred<Maybe<Cursor<LoginData>>> {
+        let cursor = ArrayCursor(data: cache.sort({ (loginA, loginB) -> Bool in
+            return loginA.hostname > loginB.hostname
+        }).map({ login in
+            return login as LoginData
+        }))
+        return Deferred(value: Maybe(success: cursor))
+    }
+
+    public func searchLoginsByUsername(username: String?, orPassword password: String?, orHostname hostname: String?) -> Deferred<Maybe<Cursor<LoginData>>> {
+        let cursor = ArrayCursor(data: cache.filter({ login in
+            var checks = [Bool]()
+            if let username = username {
+                checks.append(login.username?.startsWith(username) ?? false)
```

Comment:
I think you want `contains`, not `startsWith`. Maybe I'm looking for `@mozilla.com`?